andela/neon-ah-backend
# #162414268 Users receive verification email on signup.

#### What does this PR do?
This PR implements email verification for users on successful registration

#### Description of Task to be completed?
- Adds the functionality to send an email to the user on successful signup.
- Verifies the user when the user clicks on the `Verify Email` button in the email.

#### How should this be manually tested?
- Clone this repository on your local machine
- Run `npm install`
- Using Postman, make a post request to the sign up route : `http://localhost:3000/api/v1/auth/signup` 
using parameters in this format
     - userName: userName
     - fullName: fullName
     - email: email
     - password: password
- You should get an email containing the token
- Click on the verify email button for the User to be verified.
- Pick the token, and place it as a parameter to `http://localhost:3000/api/v1/auth/verify/` and send a POST request  
- You get a response stating that the user account has been verified.

#### What are the relevant pivotal tracker stories?
[Link to Pivotal Tracker Story](https://www.pivotaltracker.com/n/projects/2229851/stories/162414268)
